Automattic/wp-calypso
# Enhance domain-picker package encapsulation

#### Changes proposed in this Pull Request
_It includes changes from https://github.com/Automattic/wp-calypso/pull/43046 merged into it and the following._ 

This PR makes domain-picker more FSE friendly and hopefully useable anywhere. 

##### Changes are:

1. No need for external stying to look as it should
2. No more coupling with client folder
3. No more coupling with Gutenboading (S)CSS variables
4. Move as much as possible domain-picker logic to domain-picker
5. Use stores instead of props when possible (domainCategories)
6. Clean up a lot of uneeded CSS


#### Testing instructions

<s>This PR isn't testable by itself. It goes hand in hand with https://github.com/Automattic/wp-calypso/pull/43046. Please follow testing instructions there.</s> #43046 was merged into this one.
- Open /new of this branch and in production
- Domain Picker (button, modal and popover) UX should be 1:1 with no regressions
